### PR TITLE
add graph name check in gremlin and rest config

### PR DESCRIPTION
**Pull request by zhoney**
fixed: #850

Change-Id: I5c4c82c784aebeb0bfe910107d7d4e19abc62d7c

### CODE REVIEW

File: hugegraph-dist/src/main/java/com/baidu/hugegraph/cmd/InitStore.java [hunk 1]
```diff
@@ -57,6 +60,7 @@
         exceptions.put("InvalidQueryException", "unconfigured table");
     }
 
+    @SuppressWarnings("unchecked")
```
Comment (javeme) [inline hugegraph-dist/src/main/java/com/baidu/hugegraph/cmd/InitStore.java]: move to line 90

File: hugegraph-dist/src/main/java/com/baidu/hugegraph/cmd/InitStore.java [hunk 2]
```diff
@@ -83,7 +87,15 @@ public static void main(String[] args)
                         "Must contain at least one graph");
 
         for (ConfigurationNode graphName : graphNames) {
+            String name = ((Map.Entry<String, Object>)
+                           graphName.getReference()).getKey();
```
Comment (javeme) [inline hugegraph-dist/src/main/java/com/baidu/hugegraph/cmd/InitStore.java]: can we call graphName.getName()

File: hugegraph-dist/src/main/java/com/baidu/hugegraph/cmd/InitStore.java [hunk 3]
```diff
@@ -83,7 +87,15 @@ public static void main(String[] args)
...
             String configPath = graphName.getValue().toString();
+            E.checkArgument(name.matches(NAME_REGEX),
+                            "Invalid graph name '%s' in gremlin-server.yaml," +
+                            " valid graph name is up to 48 alpha-numeric " +
+                            "characters and underscores and only letters are " +
+                            "supported as first letter. " +
+                            "Note: letter is case insensitive", name);
```
Comment (javeme) [inline hugegraph-dist/src/main/java/com/baidu/hugegraph/cmd/InitStore.java]: seems better to add a method checkGraphName()

File: hugegraph-dist/src/main/java/com/baidu/hugegraph/cmd/InitStore.java (hunk 2 above)
Comment (zhoney) [reply hugegraph-dist/src/main/java/com/baidu/hugegraph/cmd/InitStore.java]: graphName.getName() returns 'graph' due to it's sub node of 'graphs'

**Pull request merged**